soft4pes/control/mpc/solvers/mpc_enum.py

```diff
@@ -0,0 +1,172 @@
+"""Enumeration based solver for Model Predictive Control (MPC)."""
+
+from itertools import product
+import numpy as np
+
+
+class MpcEnum:
+    """
+    Enumeration-based solver for Model predictive control (MPC).
+
+    Attributes
+    ----------
+    J : 1 x conv.nl^(3*Np) ndarray of floats
+        Array for costs.
+    u_seq : 3*Np x conv.nl^(3*Np) ndarray of ints
+        Array for 3-phase switch position sequences.
+    i : int
+        Counter for the switch position sequences.
+    SW_COMB : 1 x conv.nl^3 ndarray of ints
+        All possible 3-phase switch position.
+    """
+
+    def __init__(self, conv):
+        """
+        Initialize an MpcEnum instance.
+
+        Parameters
+        ----------
+        conv : converter object
+            Converter model.
+        """
+        self.J = None
+        self.u_seq = None
+        self.i = 0
+
+        if conv.nl == 2:
+            sw_pos_3ph = np.array([-1, 1])
+
+        elif conv.nl == 3:
+            sw_pos_3ph = np.array([-1, 0, 1])
+
+        # Create all possible 3-phase switch position
+        self.SW_COMB = np.array(list(product(sw_pos_3ph, repeat=3)))
+
+    def __call__(self, sys, conv, ctr, ref):
+        """
+        Solve MPC problem by using exhaustive enumeration.
+
+        Parameters
+        ----------
+        sys : system object
+            System model.
+        conv : converter object
+            Converter model.
+        ctr : controller object
+            Controller model.
+        ref : ndarray of floats
+            Reference.
+
+        Returns
+        -------
+        uk : 1 x 3 ndarray of ints
+            The 3-phase switch position sequence with the lowest cost.
+        """
+
+        # Initialize array for costs and 3-phase switch position sequences
+        self.J = np.zeros((conv.nl**(3 * ctr.Np), 1))
+        self.u_seq = np.zeros((conv.nl**(3 * ctr.Np), 3 * ctr.Np))
+        self.i = 0
+
+        self.solve(sys, conv, ctr, sys.x, ref, ctr.u_km1, 0)
```

Comment:
I would define k = 0 outside the function and then I would have k as an argument to the function. This improves the readability

Reply:
The first comment is addressed in the next commit.

I like the second one about `ell`. This would make sense, as we could differ the simulation and prediction time steps. For now, I will make a task about this, and it will be implemented later.